### PR TITLE
Attendance visual changes and template update

### PR DESCRIPTION
**Pull request by davidhtien**
create template for 24 column grid bootstrap, upgrade date-fns and fix beginning of the week functions, use icons instead of letters for leaveslip statuses, change color scheme of leaveslip details

### CODE REVIEW

**Comment by davidhtien** [comment]
@radicalray LGTM

File: ap/static/react-gulp/app/scripts/containers/GridContainer.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ EventColumn -----esr-----> EventView
 */
 
 const mapStateToProps = (state) => {
-  var weekStart = dateFns.format(dateFns.startOfWeek(state.reducer.date, {weekStartsAt: 1}), 'M/D/YY'),
-      weekEnd = dateFns.format(dateFns.endOfWeek(state.reducer.date, {weekStartsAt: 1}), 'M/D/YY');
+  var weekStart = dateFns.format(dateFns.startOfWeek(state.reducer.date, {weekStartsOn: 1}), 'M/D/YY'),
```
Comment (radicalray) [inline ap/static/react-gulp/app/scripts/containers/GridContainer.js]: Don't know if I"m reading this correctly. Isn't weekStartsOn for UI should be the current week we're on? Not sure what this actually calculates.

Comment (radicalray) [inline ap/static/react-gulp/app/scripts/containers/GridContainer.js]: Why is startson 1? What does that mean?

**Comment by radicalray** [comment]
Looks mostly good. I pulled your branched tested it. The side menu is huge and the right side is pushed to the bottom of the page...is this only ubuntu UI bug? hard refresh didn't fix the problem. Fresh install + new db everything. 

One more thing, as you scroll through periods, no ends, just keeps going haha. Also, the width of the dates are not set so as you scroll the button (left/right) jumps/shifts.

**Comment by radicalray** [comment]
Fixed most UI/CSS bugs. 

@davidhtien Periods can still go negative...is this blocked when there's a schedule?

File: ap/static/react-gulp/app/scripts/containers/GridContainer.js (hunk 1 above)
Comment (davidhtien) [inline ap/static/react-gulp/app/scripts/containers/GridContainer.js]: Means the week starts on Monday

**Comment by davidhtien** [comment]
Ton of new fixes
- weeknav disabled button looks better
- weeknav date format changed to save space
- fix api from always making a present roll
- change names of variables passed to frontend to match those in initial state file
- make posting roll and slip at the same time synchronous
- remove data-toggle from divs in roll slip form so clicking on the label also selects the appropriate radio button
- make arrows for attendance detail clickable instead of just the text
- made selecting an event open the roll and slip forms
- hide roll form in all instances from first year trainees
- limit period navigation to 0-9
- fix test data for trainee
- fix editing selected events for leave slip details

File: ap/attendance/views.py [hunk 2]
```diff
@@ -65,7 +67,7 @@ def allow_bulk_destroy(self, qs, filtered):
         # return not all(x in filtered for x in qs)
 
 class AttendanceViewSet(BulkModelViewSet):
-    queryset = Trainee.objects.all()
+    queryset = Trainee.objects.filter(is_active=True)
```
Comment (radicalray) [inline ap/attendance/views.py]: Maybe we should make is_active default in Trainee querymanager so devs won't accidentally get non-active trainees ever.

File: ap/static/react-gulp/app/scripts/index.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ const combined = combineReducers(reducers);
 
 const store = createStore(combined, initialState, compose(
   applyMiddleware(thunkMiddleware),
-  // window.devToolsExtension ? window.devToolsExtension() : f => f //redux chrome dev tools
+  window.devToolsExtension ? window.devToolsExtension() : f => f //redux chrome dev tools
```
Comment (radicalray) [inline ap/static/react-gulp/app/scripts/index.js]: Make sure to put an issue to turn this off if you intend to keep it on in production.

**Comment by radicalray** [comment]
Still has tons of errors in console:

index.js:6071 Warning: Failed propType: Required prop `submitting` was not specified in `RollSlipForm`. Check the render method of `ReduxForm(RollSlipForm)`.warning @ index.js:6071checkPropTypes @ index.js:32621validatePropTypes @ index.js:32640ReactElementValidator.createElement @ index.js:32674render @ index.js:40854ReactCompositeComponentMixin._renderValidatedComponentWithoutOwnerOrContext @ index.js:28284ReactCompositeComponentMixin._renderValidatedComponent @ index.js:28304ReactCompositeComponent__renderValidatedComponent @ index.js:34676ReactCompositeComponentMixin.performInitialMount @ index.js:27889ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342mountComponentIntoNode @ index.js:33463Mixin.perform @ index.js:38107batchedMountComponentIntoNode @ index.js:33484Mixin.perform @ index.js:38107ReactDefaultBatchingStrategy.batchedUpdates @ index.js:31510batchedUpdates @ index.js:35830ReactMount._renderNewRootComponent @ index.js:33637ReactMount__renderNewRootComponent @ index.js:34676ReactMount._renderSubtreeIntoContainer @ index.js:33714ReactMount.render @ index.js:33734React_render @ index.js:3467623../containers/Attendance @ index.js:3063s @ index.js:1e @ index.js:1(anonymous function) @ index.js:1
index.js:6071 Warning: Textarea elements must be either controlled or uncontrolled (specify either the value prop, or the defaultValue prop, but not both). Decide between using a controlled or uncontrolled textarea and remove one of these props. More info: https://fb.me/react-controlled-componentswarning @ index.js:6071ReactDOMTextarea.mountWrapper @ index.js:31107ReactDOMComponent.Mixin.mountComponent @ index.js:28983ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342mountComponentIntoNode @ index.js:33463Mixin.perform @ index.js:38107batchedMountComponentIntoNode @ index.js:33484Mixin.perform @ index.js:38107ReactDefaultBatchingStrategy.batchedUpdates @ index.js:31510batchedUpdates @ index.js:35830ReactMount._renderNewRootComponent @ index.js:33637ReactMount__renderNewRootComponent @ index.js:34676ReactMount._renderSubtreeIntoContainer @ index.js:33714ReactMount.render @ index.js:33734React_render @ index.js:3467623../containers/Attendance @ index.js:3063s @ index.js:1e @ index.js:1(anonymous function) @ index.js:1
index.js:6071 Warning: Failed propType: Required prop `roll` was not specified in `EventView`. Check the render method of `EventColumn`.warning @ index.js:6071checkPropTypes @ index.js:32621validatePropTypes @ index.js:32640ReactElementValidator.createElement @ index.js:32674(anonymous function) @ index.js:847EventColumn @ index.js:845ReactCompositeComponentMixin.mountComponent @ index.js:27775ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactMultiChild.Mixin.mountChildren @ index.js:34044ReactDOMComponent.Mixin._createInitialChildren @ index.js:29208ReactDOMComponent.Mixin.mountComponent @ index.js:29054ReactDOMComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342ReactCompositeComponentMixin.performInitialMount @ index.js:27895ReactCompositeComponentMixin.mountComponent @ index.js:27842ReactCompositeComponent_mountComponent @ index.js:34676ReactReconciler.mountComponent @ index.js:35342mountComponentIntoNode @ index.js:33463Mixin.perform @ index.js:38107batchedMountComponentIntoNode @ index.js:33484Mixin.perform @ index.js:38107ReactDefaultBatchingStrategy.batchedUpdates @ index.js:31510batchedUpdates @ index.js:35830ReactMount._renderNewRootComponent @ index.js:33637ReactMount__renderNewRootComponent @ index.js:34676ReactMount._renderSubtreeIntoContainer @ index.js:33714ReactMount.render @ index.js:33734React_render @ index.js:3467623../containers/Attendance @ index.js:3063s @ index.js:1e @ index.js:1(anonymous function) @ index.js:1
index.js:6071 Warning: Failed propType: Required prop `slip` was not specified in `EventView`. Check the render method of `EventColumn`.

Is that normal?

**Comment by davidhtien** [comment]
ah my bad, committed the debug build

**Comment by radicalray** [comment]
Additional bugs in a separate PR. Merging this one. 

Todo: 
 - [ ] Need to load TA's in dynamically
 - [ ] Fix various css/ui bugs (check box, out of alignment) open leaveslip by default
 - [ ] Fix way terms are calculated not based on absolute date but on week (calculation too fragile)
 - [ ] Better permissions for rest api in prod.py to lock down api permissions

**Pull request merged**